pkgs/stdenv/generic/setup.sh

```diff
@@ -505,6 +505,10 @@ activatePackage() {
         addToSearchPath _PATH "$pkg/bin"
     fi
 
+    if [[ "$hostOffset" -eq 0 && -d "$pkg/bin" ]]; then
+        addToSearchPath HOST_PATH "$pkg/bin"
```

Comment:
@Ericson2314 Does the name + logic look correct here? If so I might merge just this part into staging now & leave the patchshebang changes for later.

I wasn't sure whether HOST_PATH was a good name (and should we have a TARGET_PATH as well?)

Reply:
That looks good, and find with two-step plan, but can you make it depend on `strictDeps` like as I mentioned in my comment above?

Reply:
Why though? It shouldn’t be any bigger than PATH here and it’s all internal to bash so no ARG_MAX issues.

Reply:
The idea with `strictDeps = false` is things work as if we made each dependency every type of dependency, i.e. `buildInputs = nativeBuildInputs = ...`. In that case, `PATH` would indeed equal `HOST_PATH`.

Reply:
Ok but it would still be easier to have HOST_PATH always defined - less conditionals. Eventually people should start changing patchShebangs to something like patchShebangsHost & patchShebangsBuild.

Reply:
@matthewbauer oh I'm definiely agreeing always define it. In the `strictDeps = false` cause it should just contain more entries, because more pkgs's `bin` directories will count. That's all.